tests/regression/test_helmholtz_serendipity.py

```diff
@@ -0,0 +1,101 @@
+"""This demo program solves Helmholtz's equation
+
+  - div grad u(x, y) + u(x,y) = f(x, y)
+
+on the unit square with source f given by
+
+  f(x, y) = (1.0 + 8.0*pi**2)*cos(x[0]*2*pi)*cos(x[1]*2*pi)
+
+and the analytical solution
+
+  u(x, y) = cos(x[0]*2*pi)*cos(x[1]*2*pi)
+"""
+
+from os.path import abspath, dirname, join
+import numpy as np
+import pytest
+
+from firedrake import *
+
+cwd = abspath(dirname(__file__))
+
+
+def helmholtz(r, quadrilateral=True, degree=2, mesh=None):
+    # Create mesh and define function space
+    if mesh is None:
+        mesh = UnitSquareMesh(2 ** r, 2 ** r, quadrilateral=quadrilateral)
+    V = FunctionSpace(mesh, "S", degree)
+
+    x, y = SpatialCoordinate(mesh)
+
+    # Define variational problem
+    u = TrialFunction(V)
+    v = TestFunction(V)
+
+    #uex = cos(pi*x)*cos(pi*y)  # Alternative problem
+    uex = cos(x*pi*2)*cos(y*pi*2)
+    f = -div(grad(uex)) + uex
+
+    a = (inner(grad(u), grad(v)) + inner(u, v))*dx(degree=12)
+    L = inner(f, v)*dx(degree=12)
+
+    params = {"snes_type": "ksponly",
+              "ksp_type": "preonly",
+              "pc_type": "lu",
+              "snes_rtol": 1e-16,
+              "snes_atol": 1e-25}
```

Comment:
We should remove lines 45 and 46 because they aren't used when the snes_type is is-only.